### PR TITLE
Issue #1029: Print error messages if header is a folder or unreadable

### PR DESCRIPTION
**Pull request by seemyvest**
r? @fitzgen

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -1542,7 +1542,27 @@ impl Bindings {
             }
         }
 
+        #[cfg(unix)]
+        fn can_read(perms: &std::fs::Permissions) -> bool {
+            use std::os::unix::fs::PermissionsExt;
+            perms.mode() & 0o400 > 0
```
Comment (pepyakin) [inline src/lib.rs]: I'm not sure if this check is reliable enough...
The check assumes that bindgen is running under the user who owns this file which may not be the case.

Comment (fitzgen) [reply src/lib.rs]: I think `perms.mode() & 0o444 > 0` (rather than `0o400`) is Good Enough(tm)

**Comment by emilio** [comment]
@bors-servo r=fitzgen

Thanks for working on this!

**Comment by fitzgen** [comment]
Thanks @seemyvest ! Interested in finding another issue to hack on? I can help find something if nothign from the issue tracker jumps out at you :)

**Comment by durka** [comment]
Since this PR, bindgen panics with no explanation if the header does not exist. Consider not unwrapping the result of `fs::metadata`. This just came up on IRC.

**Comment by fitzgen** [comment]
Woops, I should have caught that in review! Are you interested in making a PR that fixes this @durka ? Or maybe @seemyvest ?

**Comment by seemyvest** [comment]
No problem 👍

**Pull request merged**